packages/domain/source/shared/todos/rules/unique-todo.js

```diff
@@ -0,0 +1,29 @@
+Space.domain.Rule.extend('Todos.UniqueTodoRule', {
+
+  mixin: [
+    Space.messaging.CommandSending
+  ],
+
+  enforceMap() {
+    return [{
+      'Todos.CreateTodo': this._onCreateTodo
+    }]
+  },
+
+  _onCreateTodo(command) {
+    console.log(
+      ['rule', 'before'], 'Todos.UniqueTodoRule', command.toString(),
+      ['S', 'C']
+    )
+
+    this.send(
+      new Todos.ValidateTodo({title: command.title}), function(err, result) {
+        if (result === false) {
```

Comment:
Commands are used to change state and should be ack/nak, whereas here you just need an answer to a question… therefore this should not be a command, but instead a query :wink: